service/src/main/kotlin/io/provenance/digitalcurrency/consortium/stream/EventStreamConsumer.kt

```diff
@@ -79,108 +84,128 @@ class EventStreamConsumer(
-            .forEach { event ->
+        val filteredBurns = burns.filter { it.sender.isNotEmpty() && it.recipient.isNotEmpty() }
+            .mapNotNull { event ->
                 log.debug("SC Transfer - tx: ${event.txHash} sender: ${event.sender} recipient: ${event.recipient} amount: ${event.amount} denom: ${event.denom}")
 
                 // TODO (steve) implement caching
-                // TODO (steve) move to getAttributes by tag name
-                val fromAddressBankUuid = pbcService.getAttributes(event.sender).bankUuid()
+                val fromAddressBankUuid = pbcService.getAttributeByTagName(event.sender, bankClientProperties.kycTagName)?.bankUuid()
 
                 // persist a record of this transaction if either the from or the to address has this bank's attribute
                 if (event.recipient == pbcService.managerAddress && fromAddressBankUuid != null) {
-                    // TODO (steve) change to upsert
-                    transaction {
-                        CoinMovementRecord.insert(
-                            txHash = event.txHash,
-                            fromAddress = event.sender,
-                            fromAddressBankUuid = fromAddressBankUuid,
-                            toAddress = event.recipient,
-                            toAddressBankUuid = null,
-                            blockHeight = event.height,
-                            blockTime = OffsetDateTime.parse(block.header.time),
-                            amount = event.amount,
-                            denom = event.denom,
-                            type = BURN,
-                        )
-                    }
+                    BurnWrapper(event, fromAddressBankUuid)
+                } else {
+                    null
                 }
             }
 
         // general coin transfers outside of the SC are tracked require the EventMarkerTransfer event
-        // TODO (steve) filter where denom is what we care about
-        batch.markerTransfers()
-            .filter { it.fromAddress.isNotEmpty() && it.toAddress.isNotEmpty() }
-            .forEach { event ->
+        val filteredTransfers = transfers.filter { it.fromAddress.isNotEmpty() && it.toAddress.isNotEmpty() }
+            .filter { it.denom == bankClientProperties.denom }
+            .mapNotNull { event ->
                 log.debug("MarkerTransfer - tx: ${event.txHash} from: ${event.fromAddress} to: ${event.toAddress} amount: ${event.amount} denom: ${event.denom}")
 
                 // TODO (steve) implement caching
-                // TODO (steve) move to getAttributes by tag name
-                val fromAddressBankUuid = pbcService.getAttributes(event.fromAddress).bankUuid()
-                val toAddressBankUuid = pbcService.getAttributes(event.toAddress).bankUuid()
+                val fromAddressBankUuid = pbcService.getAttributeByTagName(event.fromAddress, bankClientProperties.kycTagName)?.bankUuid()
```

Comment:
why not just grab this from the address registration table?

Reply:
Not sure haha. Although it does allow me to replicate it locally since it checks against chain instead of Jim's database.